lib/redis-bus.js

```diff
@@ -0,0 +1,59 @@
+const Redis = require('ioredis')
+
+/**
+ * This class extends the EventEmitter to act as a pub/sub message bus,
+ * allowing payloads to be received by one instance of Smee and sent by
+ * others. This allows Smee to be multi-instanced!
+ */
+module.exports = class RedisBus {
+  /**
+   * @param {import('events').EventEmitter} events
+   */
+  constructor (events) {
+    this.events = events
+
+    // If Redis isn't enabled, don't try to connect
+    if (!process.env.REDIS_URL) return
+
+    this.opts = {
+      connection: process.env.REDIS_URL,
+      namespace: 'global'
+    }
+
+    // Need two Redis clients; one cannot subscribe and publish.
+    this.sub = new Redis(this.opts.connection)
+    this.pub = new Redis(this.opts.connection)
+
+    // Subscribe to the Redis event channel
+    this.sub.subscribe(this.opts.namespace)
+
+    // When we get a message, parse it and
+    // throw it over to the EventEmitter.
+    this.sub.on('message', (_, message) => {
+      const json = JSON.parse(message)
+      return this.emitLocalEvent(json)
+    })
+  }
+
+  /**
+   * Emit an even to this machine's in-memory EventEmitter
+   */
+  emitLocalEvent ({ channel, payload }) {
+    return this.events.emit(channel, payload)
+  }
+
+  /**
+   * Emit an event to the Redis bus, which will tell every subscriber about it
+   * @param {string} channel - Channel name
+   * @param {any} payload
+   */
+  emitRemoteEvent ({ channel, payload }) {
```

Comment:
It feels a bit odd to me that the method here is `emitRemoteEvent` but can fall back to `local`. Maybe we just rename this to `emitEvent` since it can be either remote or local depending on config?

```suggestion
  emitEvent ({ channel, payload }) {
```

Reply:
That works, especially given the next comment (renaming to `EventBus` and abstracting the Redis inclusion entirely)